### PR TITLE
feat(node): fully implement `node:querystring` and `node:url`

### LINKED ISSUE #290: url.parse('/404') throws, unlike in Node.js
### Environment

Cloudflare Workers: wrangler 3.64.0 w/ `experimental:nodejs_compat_v2`

### Reproduction

`require('url').parse('/404')`

### Describe the bug

You get `TypeError: Failed to construct 'URL': Invalid URL` (or similar) due to unenv's use of `new URL()` to parse.

In Node.js, with `url.parse('/404')` you get:
```
Url {
  protocol: null,
  slashes: null,
  auth: null,
  host: null,
  port: null,
  hostname: null,
  hash: null,
  search: null,
  query: null,
  pathname: '/404',
  path: '/404',
  href: '/404'
}
```

### Additional context

This occurs in Next.js as it passes around "urls" like `/404` and `/500` internally a lot.

The [code comments note divergence is potentially an issue](https://github.com/unjs/unenv/blob/1a799443a5d3f0463531f715e22ef0b1e78b43ba/src/runtime/node/url/index.ts#L11-L13) – and indeed this is a case where it is.

The compat matrix over at https://workers-nodejs-compat-matrix.pages.dev/ should probably be changed to be a "Mismatch" instead of a "Supported" for url.parse due to this (and no doubt other strings exhibit divergence too)

### Logs

_No response_

### PR DESCRIPTION
**Pull request by pi0**
fixes #290
alternative of #292

Ported implementation of `node:url` and `node:querystring` from [Node.js v22.7.0](https://github.com/nodejs/node/tree/v22.7.0/lib).

Remarks: 
- Process was semi-automated and I tried to not touch implementation as much as possible
- Type checks were also added and passed, although some type safety (mostly for nullable values) had been bypassed trusting upstream implementation is safe
- Ported `tests/parallel/test-{querystring,url}*` tests which pass as expected
  - [ ] Invalid IDNA seems to be uncaught. possibly related to `punycode` behavior not matching native format binding
  - [ ] Whatwg URL format with unicode+auth options fails

### CODE REVIEW

File: src/runtime/node/querystring/internal/querystring.ts [hunk 1]
```diff
@@ -0,0 +1,113 @@
...
+      hexTable[0x80 | (c & 0x3f)];
+  }
+  if (lastPos === 0) return str;
+  if (lastPos < len) return out + String.prototype.slice.call(str, lastPos);
+  return out;
+}
+
+export { encodeStr };
+export { hexTable };
+export { isHexTable };
```
Comment (anonrig) [inline src/runtime/node/querystring/internal/querystring.ts:113]:
```suggestion
export { encodeStr, hexTable, isHexTable };
```

**Review by anonrig — APPROVED** [review]

File: src/runtime/node/querystring/internal/querystring.ts (hunk 1 above)
Comment (pi0) [reply src/runtime/node/querystring/internal/querystring.ts:113]: Thanks. btw this is auto generated code from unjs/nstdlib (private but you should have access)

**Comment by pi0** [comment]
dear @IgorMinar i noticed you added yourself for review. Do you have any specific concerns about this PR?

I think for next step of adopting hybrid support for workerd, we can try with another PR. (and i have two different methods i like to discuss with your team how we can do it).

**Comment by pi0** [comment]
(merging to test against nightly channel and iterate)

**Pull request merged**